### PR TITLE
Update GitHub Pages API to add new response parameters

### LINKED ISSUE #2107: GitHub Pages Response Out of Date
The GitHub Pages API response appears to be a bit out of date. There are new additional fields for GitHub Pages that aren't included in the struct.  These include information about the visibility of the page, certificate issuance info and whether or not HTTPS is enforced.  

The additional properties required are visible in the docs [here](https://docs.github.com/en/rest/reference/repos#get-a-github-pages-site) but specifically are as follows:

```json
  "public": true,
  "https_certificate": {
    "state": "approved",
    "description": "Certificate is approved",
    "domains": [
      "developer.github.com"
    ],
    "expires_at": "2021-05-22"
  },
  "https_enforced": true
```

I'm planning on working on this over the next few days and hope to have a PR soon.

### PR DESCRIPTION
**Pull request by rreichel3**
This PR updates the GitHub Pages API structs to support the latest documented response from GitHub.  

This fixes #2107 

Its based upon the documentation on GitHub's public docs [here](https://docs.github.com/en/rest/reference/repos#get-a-github-pages-site) and enables support for viewing a GitHub Pages' visibility and HTTPS configuration.

### CODE REVIEW

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:22]:
```suggestion
	HTTPSCertificate *PagesHTTPSCertificate `json:"https_certificate,omitempty"`
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:23]:
```suggestion
	HTTPSEnforced    *bool                  `json:"https_enforced,omitempty"`
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:49]:
```suggestion
// PagesHTTPSCertificate represents the HTTPS Certificate information for a GitHub Pages site.
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:50]:
```suggestion
type PagesHTTPSCertificate struct {
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:53]:
```suggestion
	Domains     []string `json:"domains,omitempty"`
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:54]:
```suggestion
	ExpiresAt   *Timestamp    `json:"expires_at,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @rreichel3!
Just a few changes, please, and then we should be ready for a second LGTM before merging.

File: github/repos_pages.go
Comment (rreichel3) [reply github/repos_pages.go]: This doesn't seem to work properly - The GitHub API is only returning a `YYYY-MM-DD` date and the `Timestamp` deserializer expects a full timestamp.  I looked at other places this library uses `Timestamp` and all of them appear to have the full date. Its a bit of a quirk with this API it seems. 

If we need it to be a timestamp we'll likely need to modify the Timestamp deserializer (which can be done, but seems a bit heavy-handed).  Happy to do whatever you think is best @gmlewis

Comment (rreichel3) [reply github/repos_pages.go]: Thanks! Dropped a comment on the Struct

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:54]:
```suggestion
	// GitHub's API doesn't return a standard Timestamp, rather it returns a YYYY-MM-DD string.
```

**Review by gmlewis — APPROVED** [review]
Thank you, @rreichel3 !
LGTM.

Awaiting second LGTM before merging.

**Comment by rreichel3** [comment]
👋 @wesleimp Just wanted to bump this!

**Review by cpanato — APPROVED** [review]
lgtm
thanks for this addition

**Comment by gmlewis** [comment]
Thank you, @cpanato !
Merging.

**Pull request merged**
